src/ScopeProvider.tsx

```diff
@@ -18,70 +26,92 @@ export const ScopeProvider = ({
   atoms: Iterable<AnyAtom>;
   children: ReactNode;
 }) => {
-  const [getParentScopedAtom] = useContext(ScopeContext);
-  const mapping = new WeakMap<AnyAtom, AnyAtom>();
-  const atomSet = new Set(atoms);
+  const getParentScopedAtom = useContext(ScopeContext);
+  const store = useStore();
+
+  if (import.meta.env?.MODE !== 'production') {
```

Comment:
This doesn't work in various environments. It's pretty hard to do it well.
It might be more troublesome than the help with the warning.